src/unix/stream.c

```diff
@@ -1001,6 +1001,7 @@ uv_handle_type uv__handle_type(int fd) {
 static void uv__stream_eof(uv_stream_t* stream, const uv_buf_t* buf) {
   stream->flags |= UV_HANDLE_READ_EOF;
   stream->flags &= ~UV_HANDLE_READING;
+  stream->flags &= ~UV_HANDLE_READABLE;
```

Comment:
how about writable?

Reply:
Does reaching the end of the input make a stream immediately non-writable? That feels false (e.g. what about `shutdown()`), although I often do end up coupling them too.